Cython/Compiler/Parsing.py

```diff
@@ -3029,8 +3037,10 @@ def p_exception_value_clause(s):
                 s.next()
         else:
             if s.sy == '?':
-                exc_check = 1
+                exc_check = True
                 s.next()
+            else:
+                exc_check = False
             exc_val = p_test(s)
```

Comment:
```suggestion
            else:
                exc_check = False
            # exc_val can be non-None even if exc_check is False, c.f. "except -1"
            exc_val = p_test(s)
```
Maybe a comment could be helpful here?  I ended up getting caught out again by the same [misapprehension](https://github.com/cython/cython/pull/4670#discussion_r817205131) that @shwina had in March, namely that it would be impossible to have `exc_check is False and exc_val is not None`